features/feature_case/cublasLt/cublaslt_macro.cu

```diff
@@ -0,0 +1,153 @@
+
+  // create a preference handle with specified max workspace
+  cublasCheck(cublasLtMatmulPreferenceCreate(&preference));
+  cublasCheck(cublasLtMatmulPreferenceSetAttribute(preference,
+      CUBLASLT_MATMUL_PREF_MAX_WORKSPACE_BYTES,
+      &cublaslt_workspace_size, sizeof(cublaslt_workspace_size)));
+
+  // find a suitable algorithm
+  cublasCheck(cublasLtMatmulAlgoGetHeuristic(cublaslt_handle, operationDesc,
+      weightLayout, inputLayout, outputLayout, outputLayout,
+      preference, 1, &heuristic, &returnedResults));
+  if (returnedResults == 0) {
+      printf("No cuBLASLt algorithm: B: %d, T: %d, C: %d, OC: %d, bias: %d, gelu: %d\n",
+          B, T, C, OC, has_bias, has_gelu);
+      exit(EXIT_FAILURE);
+  }
+
+  // call the matmul
+  const float alpha = 1.0f, beta = 0.0f;
+  cublasCheck(cublasLtMatmul(cublaslt_handle, operationDesc,
+      &alpha, weight, weightLayout, inp, inputLayout, &beta,
+      out, outputLayout, out, outputLayout, &heuristic.algo,
+      cublaslt_workspace, cublaslt_workspace_size, 0));
+
+  // cleanups
+  cublasCheck(cublasLtMatmulPreferenceDestroy(preference));
+  cublasCheck(cublasLtMatmulDescDestroy(operationDesc));
+  cublasCheck(cublasLtMatrixLayoutDestroy(weightLayout));
+  cublasCheck(cublasLtMatrixLayoutDestroy(inputLayout));
+  cublasCheck(cublasLtMatrixLayoutDestroy(outputLayout));
+  cublasCheck(cublasLtMatrixLayoutDestroy(biasLayout));
+}
+
+
+void matmul_forward(float* out,
+                    const float* inp, const float* weight, const float* bias,
+                    int B, int T, int C, int OC){
+                    
+                    
+  matmul_forward_cublaslt(out, inp, weight, bias, B, T, C, OC);                    
+
+}
+
+int main(int argc, char **argv) {
+    srand(0);
+
+    int B = 32;
+    int T = 1024;
+    int C = 768;
+    int OC = 768 * 4; // expansion of 4, e.g. in the MLP
+
+    // set up the device
+    int deviceIdx = 0;
+    cudaCheck(cudaSetDevice(deviceIdx));
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, deviceIdx);
+    printf("Device %d: %s\n", deviceIdx, deviceProp.name);
+
+    // setup cuBLAS and cuBLASLt
+    cublasCheck(cublasCreate(&cublas_handle));
+    cublasCheck(cublasLtCreate(&cublaslt_handle));
+    // TF32 precision is equivalent to torch.set_float32_matmul_precision('high')
+    int enable_tf32 = deviceProp.major >= 8 ? 1 : 0;
+    printf("enable_tf32: %d\n", enable_tf32);
+    cublas_compute_type = enable_tf32 ? CUBLAS_COMPUTE_32F_FAST_TF32 : CUBLAS_COMPUTE_32F;
+    cublasMath_t cublas_math_mode = enable_tf32 ? CUBLAS_TF32_TENSOR_OP_MATH : CUBLAS_DEFAULT_MATH;
+    cublasCheck(cublasSetMathMode(cublas_handle, cublas_math_mode));
+    // setup the (global) cuBLASLt workspace
+    cudaCheck(cudaMalloc(&cublaslt_workspace, cublaslt_workspace_size));
+
+    // create host memory of random numbers
+    float* out = (float*)malloc(B * T * OC * sizeof(float));
+    float* inp = make_random_float(B * T * C);

```

Comment:
1. there is no definitaion of `make_random_float`.
2. I prefer using fixed input than using random input. Fixed input => fixed output is more convenient to compare the result in cuda and sycl.